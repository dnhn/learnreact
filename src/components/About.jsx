import React, { PureComponent, createRef } from 'react';
import { connect } from 'react-redux';
import { closeAbout } from '../redux/actions';

import './About.scss';

export class About extends PureComponent {
  constructor(props) {
    super(props);
    this.closeAboutBtn = createRef();
  }

  componentDidMount() {
    this.closeAboutBtn.current.focus();
  }

  closeAbout = e => e.target === this.thisAbout && this.props.closeAbout();

  render = () => (
    <div
      ref={ref => this.thisAbout = ref}
      className="about"
      onClick={e => this.closeAbout(e)}
    >
      <div className="about__popup">
        <h1>React Photo Grid</h1>
        <p>
          <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">React.js</a> practise by <a href="https://github.com/nhantdn" target="_blank" rel="noopener noreferrer">Nhan</a>
        </p>
        <p>
          <a href="https://www.netlify.com" target="_blank" rel="noopener noreferrer">
            <img src="https://www.netlify.com/img/global/badges/netlify-dark.svg" alt="Deploys by Netlify" />
          </a>
        </p>
        <button
          ref={this.closeAboutBtn}
          type="button"
          className="about__close"
          onClick={this.props.closeAbout}
        >
          Close
        </button>
      </div>
    </div>
  );
}

const mapDispatchToProps = dispatch => ({
  closeAbout: () => dispatch(closeAbout()),
});

export default connect(null, mapDispatchToProps)(About);
