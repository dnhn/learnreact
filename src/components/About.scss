.about {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  padding: 5%;
  background-color: rgba(0, 0, 0, .5);
  text-align: center;
}

.about__popup {
  margin: auto;
  padding: 3%;
  border-radius: 4px;
  box-shadow: 0 10px 10px -10px black;
  background-color: white;
}

.about__close {
  border: 1px solid #333;
  border-radius: 4px;
  padding: 2% 5%;

  &:hover,
  &:focus,
  &:active {
    background-color: black;
    color: white;
  }
}

@media screen and (max-width: 767px) {
  .about {
    display: flex;
  }

  .about__popup {
    padding: 5%;
  }
}

@media screen and (min-width: 768px) {
  .about__popup {
    width: 40vw;
  }
}
