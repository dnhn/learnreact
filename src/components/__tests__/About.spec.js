import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { About } from '../About';

const mockStore = {
  closeAbout: jest.fn(),
}

describe('<About />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('closes on click', () => {
    const { getByText } = render(<About {...mockStore} />);
    fireEvent.click(getByText('Close'));
    expect(mockStore.closeAbout).toHaveBeenCalled();
  });

  it('closes on backdrop click', () => {
    const { container } = render(<About {...mockStore} />);
    fireEvent.click(container.getElementsByClassName('about')[0]);
    expect(mockStore.closeAbout).toHaveBeenCalled();
  });
});
