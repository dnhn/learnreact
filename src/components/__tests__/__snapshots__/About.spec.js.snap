// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="about"
  >
    <div
      class="about__popup"
    >
      <h1>
        React Photo Grid
      </h1>
      <p>
        <a
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          React.js
        </a>
         practise by 
        <a
          href="https://github.com/nhantdn"
          rel="noopener noreferrer"
          target="_blank"
        >
          Nhan
        </a>
      </p>
      <p>
        <a
          href="https://www.netlify.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Deploys by Netlify"
            src="https://www.netlify.com/img/global/badges/netlify-dark.svg"
          />
        </a>
      </p>
      <button
        class="about__close"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
